utils/src/main/java/software/amazon/awssdk/utils/cache/NonBlocking.java

```diff
@@ -79,10 +150,35 @@ public void initializeCachedSupplier(CachedSupplier<?> cachedSupplier) {
     }
 
     private void scheduleRefresh(CachedSupplier<?> cachedSupplier) {
-        executor.schedule(() -> {
+        SCHEDULER.schedule(() -> {
+            Thread.currentThread().setName(asyncThreadName);
+
+            if (shutdown) {
+                return;
+            }
+
+            if (!BACKGROUND_REFRESH_LEASE.tryAcquire()) {
+                log.warn(() -> "Skipped a background refresh to limit SDK resource consumption. Are you closing your SDK "
```

Comment:
Because this is a background thread, there's nowhere to throw an exception to. 

If the concern is too much logging, this should really never occur unless something is seriously wrong with what the customer is doing. At that point, the worst-case scenario is that it'll log once per minute, per resource (probably credential provider). We could build something to limit the logging further, but in this "something is seriously wrong" scenario we will already be logging a lot of other stuff in other parts of the SDK, so this is just a drop in the bucket.